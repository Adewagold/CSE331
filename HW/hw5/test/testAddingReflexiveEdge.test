# Test MarvelPaths doesn't allow adding reflexive edge

# Parse through the data and build the graph
LoadGraph three_nodes_graph threeNodesWithReflexiveEdge.tsv

# List nodes and edges to see if the graph is built correctly
ListNodes three_nodes_graph
ListChildren three_nodes_graph a
ListChildren three_nodes_graph b
ListChildren three_nodes_graph c

# Find the shortest path a to c
FindPath three_nodes_graph a c

# Find the shortest path c to a
FindPath three_nodes_graph c a