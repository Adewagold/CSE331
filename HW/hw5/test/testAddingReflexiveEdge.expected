# Test MarvelPaths doesn't allow adding reflexive edge

# Parse through the data and build the graph
loaded graph three_nodes_graph

# List nodes and edges to see if the graph is built correctly
three_nodes_graph contains: a b c
the children of a in three_nodes_graph are: b(AB)
the children of b in three_nodes_graph are: a(AB) c(BC)
the children of c in three_nodes_graph are: b(BC)

# Find the shortest path a to c
path from a to c:
a to b via AB
b to c via BC

# Find the shortest path c to a
path from c to a:
c to b via BC
b to a via AB